config/tailwind/forms.js

```diff
@@ -0,0 +1,84 @@
+import plugin from 'tailwindcss/plugin'
+
+export default plugin(({ addComponents }) => {
+  // Base styles
+  const inputs = {
+    '.field-group': {
+      '@apply flex flex-col gap-8': {},
+      // label container
+      '& label': {
+        '@apply flex flex-col [&:has(.sr-only:last-child)]:sr-only': {},
+      },
+      // label
+      '& .field-label': {
+        '@apply text-gray-900 dark:text-white': {},
+      },
+      // sublabel & instructions
+      '& .field-sublabel, & .field-instructions': {
+        '@apply text-sm text-gray-700 dark:text-gray-300': {},
+      },
+      // core field styles
+      '& input[type="text"], & input[type="password"], & input[type="email"], & input[type="number"], & input[type="tel"], & input[type="url"], & select, & textarea':
+        {
+          // base styles
+          '@apply rounded-sm border-0 bg-white ring-1 outline-none ring-black/50 px-8 dark:bg-white/10 dark:text-white dark:ring-white/50':
+            {},
+          // placeholder styles
+          '@apply placeholder:text-black/40 dark:placeholder:text-white/50': {},
+          // hover styles
+          '@apply hover:ring-black/75 dark:hover:ring-white/75': {},
+          // focus styles
+          '@apply focus:ring-blue-500 focus:ring-2 dark:focus:ring-white/75':
+            {},
+          // disabled styles
+          '@apply disabled:bg-gray-500/10 disabled:ring-gray-500/20 disabled:text-black/75 disabled:cursor-not-allowed dark:disabled:ring-white/20 dark:disabled:bg-white/25 dark:disabled:text-white/50 dark:disabled:placeholder:text-white/35':
+            {},
+          // Exclude textarea from fix height
+          '&:not(textarea)': {
+            '@apply h-40': {},
+          },
+          '&:is(input[type="checkbox"])': {
+            '@apply size-20 border-0 text-blue-500 checked:bg-blue-500 min-h-0':
+              {},
+          },
+          '&:is([aria-invalid="true"])': {
```

Comment:
https://github.com/vigetlabs/blueprint/blob/main/config/tailwind/inputs.js#L65

The error state slightly deviates from Blueprint here. 

It's worth setting aria-invalid on inputs with errors... so this lets us only modify a single attribute rather than keeping both `data-state` and `aria-invalid` in sync.